hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/exception/S3ErrorTable.java

```diff
@@ -126,6 +126,10 @@ private S3ErrorTable() {
   public static final OS3Exception NO_OVERWRITE = new OS3Exception(
       "Conflict", "Cannot overwrite file with directory", HTTP_CONFLICT);
 
+  public static final OS3Exception INVALID_LENGTH = new OS3Exception(
+      "InvalidLength", "Provided length did not match the actual body length." +
+      " The content might have been changed mid-way.", HTTP_BAD_REQUEST);
```

Comment:
According to [AWS S3 error codes](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList) the return code and message for this scenario should be

`IncompleteBody | You did not provide the number of bytes specified by the Content-Length HTTP header. | 400 Bad Request `